### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by kylehuang24**

### CODE REVIEW

File: scroller.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    # Get scroll height
+    last_height = driver.execute_script("return document.body.scrollHeight")
+
+    while True:
+        # Scroll down to bottom
+        driver.execute_script("window.scrollTo(0, document.body.scrollHeight);")
+
+        # Wait to load page
+        time.sleep(SCROLL_PAUSE_TIME)
```
Comment (DanMeyer) [inline scroller.py:14]: Why a `sleep` versus some kind of "wait until the page has loaded"?

File: batch.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
+import argparse
+
+def command_line():
```
Comment (DanMeyer) [inline batch.py:3]: I think you could do this where you have a tuple/list of all the batches:
```
batches = (
  ("S21", "Summer21"),
  ("W21", "Winter21"),
)
```

and then loop through those to do `parser.add_argument` and the `if args...`

File: batch.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    parser.add_argument('-W06', '--Winter06', action='store_true', 
+        help="shows output")
+    parser.add_argument('-S06', '--Summer06', action='store_true', 
+        help="shows output")
+    parser.add_argument('-S05', '--Summer05', action='store_true', 
+        help="shows output")
+    args = parser.parse_args()
+    
+    string = "?"
+    if args.Winter21:
```
Comment (DanMeyer) [inline batch.py:76]: I _think_ you can say:
`if "Winter21" in args`

so you can do something like
```
for (full_name, short_name) in batches:
  if full_name in args:
   # probably should use f strings here
    s = s + "batch=" + short_name + "&"
```

File: scraper.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    search_query = 'https://www.ycombinator.com/companies/' + batch.command_line()
+
+    driver = webdriver.Chrome(executable_path='C:\\Users\\Kyle\\Downloads\\chromedrivers\\chromedriver.exe')
+    company_information = []
+    founder_information = []
+    driver.get(search_query)
+    time.sleep(5)
+
+    # Scroll to the bottom of page
+    # scroller.scroll(driver)
```
Comment (DanMeyer) [inline scraper.py:18]: Was commented-out for demo

File: scraper.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    company_counter = 1
+
+    for each_company in company_list:
+        # Getting company info
+        company_name = each_company.find_elements_by_class_name("styles-module__coName___3zz21")[0]
+        company_location = each_company.find_elements_by_class_name("styles-module__coLocation___yhKam")[0]
+        company_YC_batch = each_company.find_elements_by_class_name("styles-module__pill___1Cdn5")[0]
+        company_industries_array.append(industries(each_company, company_counter))
+        company_counter += 1
+        # Get information from new tab
```
Comment (DanMeyer) [inline scraper.py:31]: Just for readability, I'd add a space here (and before all the other comment lines)

Comment (DanMeyer) [reply scraper.py:31]: (Also, I dig that there are so many explanatory comments :)

File: scraper.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        company_info = [company_name.text, company_location.text, company_YC_profile, company_domain, company_YC_batch.text, ", ".join(company_industries_array[company_counter - 2])]
+        # Saving into company_information
+        company_information.append(company_info)
+        founder_information.append(information_from_new_tab[2])
+    driver.quit()
+
+    company_information_df = pd.DataFrame(company_information)
+    company_information_df.columns = ['Company', 'Location', 'YC Profile Page', 'Domain Name', 'YC Batch', 'Industries']
+    founder_information_df = pd.DataFrame(founder_information)
+
```
Comment (DanMeyer) [inline scraper.py:44]: Extension idea:

I wonder if there's a way to save and restore the data frame if the scraper gets interrupted? Or maybe write to the files intermittently so progress is bookmarked? (But then you'd need a bunch more code to read/write the files, so maybe not worth it for now)

Comment (DanMeyer) [reply scraper.py:44]: As discussed verbally, I think this is too much work for the benefit right now. So you can just resolve this comment.

**Review by DanMeyer — CHANGES_REQUESTED** [review]
LGTM. Just a few small notes in comments.

File: batch.py (hunk 2 above)
Comment (kylehuang24) [reply batch.py:3]: I played around with this. I'm just not sure you can add an argument with a variable name. It keeps saying it is an unrecognized argument when I pass the variable as the parameter in add_argument. Maybe it is a problem with variable assignment? Or maybe argparse doesn't allow you to pass variables as a parameter in add_argument?

https://docs.python.org/3/library/argparse.html#name-or-flags

**Pull request merged**